File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/navigation/Destination.java
package org.firstinspires.ftc.teamcode.navigation;

import java.util.Arrays;

/**
 * From IsaacArrayDemo 2/6/21
 */
public class Destination {

    //declare variables
    public String destName;
    private float[] coordinate;

    //constructors of all shapes and sizes.
    public Destination() {
        coordinate = new float[2];
        Arrays.fill(coordinate, 0);
    }
    public Destination(String name, float destX, float destY){
        this();
        destName = name;
        coordinate[0] = destX;
        coordinate[1] = destY;
    }


    //methods
    public String getName(){
        return destName;
    }
    public void setName(String name) {
        destName = name;
    }
    public float getX() {
        return coordinate[0];
    }
    public float getY() {
        return coordinate[1];
    }

    public void setX(float x) {
        coordinate[0] = x;
    }
    public void setY(float y) {
        coordinate[1] = y;
    }

} //END:  class Destination

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/pathfinder/Destination.kt
package org.firstinspires.ftc.teamcode.util.pathfinder

import org.firstinspires.ftc.robotcore.external.matrices.OpenGLMatrix

class Destination(val name: String, val destX: Float, val destY: Float, val clearanceRadius: Float) {
    //methods
    //declare variables
    // private val coordinates: FloatArray = FloatArray(2)
    // val x: Float
        // get() = coordinates[0]
    // val y: Float
        // get() = coordinates[1]
    val matrix: OpenGLMatrix
        get() = OpenGLMatrix.translation(destX, destY, 0F)

    // constructor
    // init {
        // coordinates[0] = destX
        // coordinates[1] = destY
    // }
}